SIGSEGV while importing vsphere_host_port_group

### Community Guidelines

- [x] I have read and agree to the [HashiCorp Community Guidelines ](https://www.hashicorp.com/community-guidelines).
- [x] Vote on this issue by adding a 👍 [reaction](https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue initial description to help the maintainers prioritize.
- [x] Do not leave "+1" or other comments that do not add relevant information or questions.
- [x] If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### Terraform

OpenTofu v1.9.0

### Terraform Provider

v2.11.1

### VMware vSphere

8.0.3.00200

### Description

When attempting to `import vsphere_host_port_group.this["192.0.0.24"]`, I get the following error:
```
vsphere_host_port_group.this["192.0.0.24"]: Importing from ID "tf-HostPortGroup:host-1020:(10)My-Network"...
vsphere_host_port_group.this["192.0.0.24"]: Import prepared!
  Prepared vsphere_host_port_group for import
vsphere_host_port_group.this["192.0.0.24"]: Refreshing state... [id=tf-HostPortGroup:host-1020:(10)My-Network]
╷
│ Error: Plugin did not respond
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
╵


Stack trace from the terraform-provider-vsphere plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x1013d7680]

goroutine 139 [running]:
github.com/hashicorp/terraform-provider-vsphere/vsphere.flattenHostNetworkSecurityPolicy(0x140001adc80, 0x0)
        /home/runner/work/terraform-provider-vsphere/terraform-provider-vsphere/provider/vsphere/host_network_policy_structure.go:223 +0x20
github.com/hashicorp/terraform-provider-vsphere/vsphere.flattenHostNetworkPolicy(0x140001adc80, 0x1400062a440)
        /home/runner/work/terraform-provider-vsphere/terraform-provider-vsphere/provider/vsphere/host_network_policy_structure.go:275 +0x2c
github.com/hashicorp/terraform-provider-vsphere/vsphere.flattenHostPortGroupSpec(0x140001adc80, 0x1400062a418)
        /home/runner/work/terraform-provider-vsphere/terraform-provider-vsphere/provider/vsphere/host_port_group_structure.go:65 +0x90
github.com/hashicorp/terraform-provider-vsphere/vsphere.resourceVSphereHostPortGroupRead(0x140001adc80, {0x1018a76a0?, 0x1400061d470})
        /home/runner/work/terraform-provider-vsphere/terraform-provider-vsphere/provider/vsphere/resource_vsphere_host_port_group.go:96 +0x118
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x101c07d78?, {0x101c07d78?, 0x140001cab70?}, 0xd?, {0x1018a76a0?, 0x1400061d470?})
        /home/runner/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.35.0/helper/schema/resource.go:811 +0x130
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0x1400002cd20, {0x101c07d78, 0x140001cab70}, 0x140005a85b0, {0x1018a76a0, 0x1400061d470})
        /home/runner/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.35.0/helper/schema/resource.go:1117 +0x408
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0x140000fdce0, {0x101c07d78?, 0x140001ca660?}, 0x14000a09b40)
        /home/runner/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.35.0/helper/schema/grpc_provider.go:710 +0x5b8
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0x14000402d20, {0x101c07d78?, 0x140007976b0?}, 0x14000056070)
        /home/runner/go/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.25.0/tfprotov5/tf5server/server.go:783 +0x228
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0x101bc3f80, 0x14000402d20}, {0x101c07d78, 0x140007976b0}, 0x140001ad180, 0x0)
        /home/runner/go/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.25.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:575 +0x1c0
google.golang.org/grpc.(*Server).processUnaryRPC(0x140001a8e00, {0x101c07d78, 0x14000797620}, {0x101c0ce80, 0x140005e8000}, 0x14000644000, 0x140001f9aa0, 0x1026e2e28, 0x0)
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.67.1/server.go:1394 +0xb78
google.golang.org/grpc.(*Server).handleStream(0x140001a8e00, {0x101c0ce80, 0x140005e8000}, 0x14000644000)
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.67.1/server.go:1805 +0xb20
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.67.1/server.go:1029 +0x8c
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 55
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.67.1/server.go:1040 +0x13c

Error: The terraform-provider-vsphere plugin crashed!
```

### Affected Resources or Data Sources

resource/vpshere_host_port_group

### Terraform Configuration

```
resource "vsphere_host_port_group" "this" {
  for_each = module.vsphere_data.vsphere_hosts
  name = var.network_name
  host_system_id = each.value.id
  virtual_switch_name = var.virtual_switch_name
  vlan_id = var.vlan_id
}
```

### Debug Output

```
╷
│ Error: Plugin did not respond
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
╵


Stack trace from the terraform-provider-vsphere plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x1013d7680]

goroutine 139 [running]:
github.com/hashicorp/terraform-provider-vsphere/vsphere.flattenHostNetworkSecurityPolicy(0x140001adc80, 0x0)
        /home/runner/work/terraform-provider-vsphere/terraform-provider-vsphere/provider/vsphere/host_network_policy_structure.go:223 +0x20
github.com/hashicorp/terraform-provider-vsphere/vsphere.flattenHostNetworkPolicy(0x140001adc80, 0x1400062a440)
        /home/runner/work/terraform-provider-vsphere/terraform-provider-vsphere/provider/vsphere/host_network_policy_structure.go:275 +0x2c
github.com/hashicorp/terraform-provider-vsphere/vsphere.flattenHostPortGroupSpec(0x140001adc80, 0x1400062a418)
        /home/runner/work/terraform-provider-vsphere/terraform-provider-vsphere/provider/vsphere/host_port_group_structure.go:65 +0x90
github.com/hashicorp/terraform-provider-vsphere/vsphere.resourceVSphereHostPortGroupRead(0x140001adc80, {0x1018a76a0?, 0x1400061d470})
        /home/runner/work/terraform-provider-vsphere/terraform-provider-vsphere/provider/vsphere/resource_vsphere_host_port_group.go:96 +0x118
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x101c07d78?, {0x101c07d78?, 0x140001cab70?}, 0xd?, {0x1018a76a0?, 0x1400061d470?})
        /home/runner/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.35.0/helper/schema/resource.go:811 +0x130
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0x1400002cd20, {0x101c07d78, 0x140001cab70}, 0x140005a85b0, {0x1018a76a0, 0x1400061d470})
        /home/runner/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.35.0/helper/schema/resource.go:1117 +0x408
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0x140000fdce0, {0x101c07d78?, 0x140001ca660?}, 0x14000a09b40)
        /home/runner/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.35.0/helper/schema/grpc_provider.go:710 +0x5b8
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0x14000402d20, {0x101c07d78?, 0x140007976b0?}, 0x14000056070)
        /home/runner/go/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.25.0/tfprotov5/tf5server/server.go:783 +0x228
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0x101bc3f80, 0x14000402d20}, {0x101c07d78, 0x140007976b0}, 0x140001ad180, 0x0)
        /home/runner/go/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.25.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:575 +0x1c0
google.golang.org/grpc.(*Server).processUnaryRPC(0x140001a8e00, {0x101c07d78, 0x14000797620}, {0x101c0ce80, 0x140005e8000}, 0x14000644000, 0x140001f9aa0, 0x1026e2e28, 0x0)
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.67.1/server.go:1394 +0xb78
google.golang.org/grpc.(*Server).handleStream(0x140001a8e00, {0x101c0ce80, 0x140005e8000}, 0x14000644000)
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.67.1/server.go:1805 +0xb20
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.67.1/server.go:1029 +0x8c
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 55
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.67.1/server.go:1040 +0x13c

Error: The terraform-provider-vsphere plugin crashed!
```

### Panic Output

_No response_

### Expected Behavior

vsphere_host_port_group is imported into state

### Actual Behavior

terraform-provider-vsphere plugin crashes

### Steps to Reproduce

1. Create a Host Port Group on a standard vSwitch on a host
2. Attempt to import that network

### Environment Details

_No response_

### Screenshots

_No response_

### References

_No response_

--- a/vsphere/host_network_policy_structure.go
+++ b/vsphere/host_network_policy_structure.go
@@ -272,14 +272,20 @@ func expandHostNetworkPolicy(d *schema.ResourceData) *types.HostNetworkPolicy {
 // flattenHostNetworkPolicy reads various fields from a HostNetworkPolicy into
 // the passed in ResourceData.
 func flattenHostNetworkPolicy(d *schema.ResourceData, obj *types.HostNetworkPolicy) error {
-	if err := flattenHostNetworkSecurityPolicy(d, obj.Security); obj.Security != nil && err != nil {
-		return err
+	if obj.Security != nil {
+		if err := flattenHostNetworkSecurityPolicy(d, obj.Security); err != nil {
+			return err
+		}
 	}
-	if err := flattenHostNicTeamingPolicy(d, obj.NicTeaming); obj.NicTeaming != nil && err != nil {
-		return err
+	if obj.NicTeaming != nil {
+		if err := flattenHostNicTeamingPolicy(d, obj.NicTeaming); err != nil {
+			return err
+		}
 	}
-	if err := flattenHostNetworkTrafficShapingPolicy(d, obj.ShapingPolicy); obj.ShapingPolicy != nil && err != nil {
-		return err
+	if obj.ShapingPolicy != nil {
+		if err := flattenHostNetworkTrafficShapingPolicy(d, obj.ShapingPolicy); err != nil {
+			return err
+		}
 	}
 	return nil
 }

